### PR TITLE
support new imc config in yml

### PR DESCRIPTION
**Pull request by minweipan**

### CODE REVIEW

**Comment by minweipan** [comment]
add new parameters for imc yml config, please review the changes, thanks @weia5-emc @XiaowenJiang

File: infrasim/model/elements/pci_imc.py [hunk 1]
```diff
@@ -37,13 +37,17 @@ def init(self):
         self.id = self.__imc_info["id"]
         self.bus = self.__imc_info["bus"]
         self.addr = self.__imc_info.get("addr")
+        self.imc_slot_topo = self.__imc_info.get("imc_slot_topo")
+        self.imc_cpu_index = self.__imc_info.get("imc_cpu_index")
```
Comment (XiaowenJiang) [inline infrasim/model/elements/pci_imc.py]: Are these two arguments optional or mandatory?
If they are optional, you should check if they have values in handle_parms before inserting them into args.

**Comment by XiaowenJiang** [comment]
retest this please

File: infrasim/model/elements/pci_imc.py (hunk 1 above)
Comment (minweipan) [reply infrasim/model/elements/pci_imc.py]: Yes, they are mandatory for warnado platform.  
Precheck function code updated.

File: infrasim/model/elements/pci_imc.py [hunk 2]
```diff
@@ -37,13 +43,17 @@ def init(self):
         self.id = self.__imc_info["id"]
         self.bus = self.__imc_info["bus"]
         self.addr = self.__imc_info.get("addr")
+        self.imc_slot_topo = self.__imc_info.get("imc_slot_topo")
```
Comment (weia5-emc) [inline infrasim/model/elements/pci_imc.py]: missing declaration in __init__() function. 
this is unsafe.

Comment (minweipan) [reply infrasim/model/elements/pci_imc.py]: code updated, please help to check the latest changes, thanks

**Comment by weia5-emc** [comment]
retest this please

**Comment by chenge3** [comment]
retest this please

**Comment by emcinfrasim** [comment]
<details><summary>Test Error Details</summary><p>
TestName: test_set_drive_page_file_exist 
<br>ErrorType: exceptions.AssertionError 
<br>ErrorMessage: line None, 
-------------------- >> begin captured logging << --------------------
0Model: INFO: Fill OEM json file
0Model: INFO: oem_data.json not exist or nvme_map_dict is empty
--------------------- >> end captured logging << --------------------- 
<br> 
<br></p></details>

**Comment by emcinfrasim** [comment]
let #508 pass and merge then retest this one.

**Comment by emcinfrasim** [comment]
retest this please

**Pull request merged**